### PR TITLE
Add docstring note about using `pkgdir` and `pkgversion` on the current module

### PR DESCRIPTION
**Pull request by IanButterworth**
Updated to just docstring examples based on https://github.com/JuliaLang/julia/pull/45695#discussion_r898345627

### CODE REVIEW

File: base/loading.jl [hunk 1]
```diff
@@ -425,32 +427,41 @@ julia> pkgdir(Foo)
...
+!!! compat "Julia 1.9"
+    The current module (no Module argument) methods require at least Julia 1.9.
 """
 function pkgdir(m::Module, paths::String...)
     rootmodule = moduleroot(m)
     path = pathof(rootmodule)
     path === nothing && return nothing
     return joinpath(dirname(dirname(path)), paths...)
 end
+pkgdir() = pkgdir(@__MODULE__)
```
Comment (KristofferC) [inline base/loading.jl:447]: This will just immediately be macro expanded to `Base`, or? Don't you need a macro for this type of functionality?

Comment (IanButterworth) [reply base/loading.jl:447]: oh, true. This could either become macro versions, or just a note in the docstrings that `pkgdir(@__MODULE__)` etc. works

Comment (IanButterworth) [reply base/loading.jl]: updated

**Review by KristofferC — APPROVED** [review]

**Comment by IanButterworth** [comment]
@giordano pointed out that moving the `@__MODULE__` to an arg defaut value would make it evaluate in the target module, so the original idea could be brought back. But I'll save that for another PR as I think this gets a lot of the way.

**Pull request merged**